### DISCUSSION TITLE
TypeError: option expires is invalid

### DISCUSSION THREAD

**Discussion by SawkaDev** [nextauthjs/next-auth#7922] [Help]
### Question 💬

I am deploying a Nextjs app using Next-Auth to AWS App Runner. I am using the Cognito Provider. 

In the logs I can see the following error: 

```
2023-06-29T23:08:26.275-04:00	TypeError: option expires is invalid
	2023-06-29T23:08:26.275-04:00	at serialize (/app/node_modules/cookie/index.js:161:13)
	2023-06-29T23:08:26.275-04:00	at setCookie (/app/node_modules/next-auth/next/utils.js:26:46)
	2023-06-29T23:08:26.275-04:00	at /app/node_modules/next-auth/next/index.js:36:147
```

I have tried everything to get rid of this. This issue does not occur on localhost. 

### How to reproduce ☕️

This is my [...nextauth].ts file

```
export function getOptions(req: NextApiRequest): NextAuthOptions {
  return {
    // @ts-ignore
    adapter: MongoDBAdapter(clientPromise as any),
    secret: process.env.NEXTAUTH_SECRET_KEY,
    providers: [
      CognitoProvider({
        name: 'Cognito',
        clientId: process.env.COGNITO_CLIENT_ID!,
        clientSecret: process.env.COGNITO_CLIENT_SECRET!,
        issuer: `https://cognito-idp.${process.env.COGNITO_REGION}.amazonaws.com/${process.env.COGNITO_POOL_ID}`
      })
    ],
    session: {
      strategy: 'jwt',
      maxAge: Number(process.env.JWT_TIMEOUT!)
    },
    jwt: {
      secret: process.env.JWT_SECRET
    },
    pages: {
      signIn: '/login',
      newUser: '/register',
      verifyRequest: '/verify-request'
    }
  };
}

var nextAuth = (req: NextApiRequest, res: NextApiResponse) => NextAuth(req, res, getOptions(req));
export default nextAuth;
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by SawkaDev** [comment]
Removing       maxAge: Number(process.env.JWT_TIMEOUT!) fixed the issue for now.

**Reply by jack-szeto** [reply]
how could this be an issue... smh

**Comment by fedefrei** [comment]
I found out a workaround hardcoding the maxAge number doing this:

```
session: {
	maxAge: 3600,
},
```

instead of 

```
session: {
	maxAge: Number(process.env.NEXTAUTH_SESSION_AGE),
},
```

if its of any help.

**Comment by jack-szeto** [comment]
Okay, I just did some checking and logging. Turns out that's something wrong in the `.env`. 
In my case, I put a comment message after the number. And it loads everything.

```.env
NEXTAUTH_JWT_TIMEOUT=86400 # for 1 day - 86400 = 1* 24 * 60 * 60
```
```ts
process.env.NEXTAUTH_JWT_TIMEOUT // "86400 # for 1 day - 86400 = 1* 24 * 60 * 60"
```

So, I just need to put the comment above the declaration.
 
```.env
# for 1 day - 86400 = 1* 24 * 60 * 60
NEXTAUTH_JWT_TIMEOUT=86400
```

![WillSmithWillSmithSmhGIF](https://github.com/nextauthjs/next-auth/assets/47553851/28fdc5b2-07d5-433d-8b7e-15dd36b47d19)

**Comment by tunafysh** [comment]
It's also a possibility that your postgresql database uses ```DATE``` for expiry instead of ```TIMESTAMP```. This is what triggered the error for me.
